src/rt/cover.d

```diff
@@ -149,10 +149,75 @@ uint parseNum(const(char)[] s)
 T min(T)(T a, T b) { return a < b ? a : b; }
 T max(T)(T a, T b) { return b < a ? a : b; }
 
+version (unittest) __gshared bool inUnittest;
+
+// Parse a string looking like a boolean and assign the result to the global
+// merge variable.
+string parseMergeOpt(string opt) @nogc nothrow
+{
+    switch (opt)
+    {
+        case "no", "No", "NO", "n", "N", "0":
+            merge = false;
+            return "0";
+        case "", "yes", "Yes", "YES", "y", "Y", "1":
+            merge = true;
+            return "1";
+        default:
+            version (unittest) if (inUnittest) return "err";
+            fprintf(stderr, "DRT Error: Runtime option 'covmerge' " ~
+                    "should be 1/y/yes or 0/n/no, got '%.*s'.\n",
+                    cast(int)opt.length, opt.ptr);
+            return "err";
+    }
+}
+
+unittest
+{
+    inUnittest = true;
+    scope (exit) inUnittest = false;
+
+    bool old_merge = merge;
+    scope (exit) merge = old_merge;
+
+    merge = true;
+    assert(parseMergeOpt("0") == "0" && !merge);
+    assert(parseMergeOpt("1") == "1" && merge);
+    assert(parseMergeOpt("n") == "0" && !merge);
+    assert(parseMergeOpt("y") == "1" && merge);
+    assert(parseMergeOpt("N") == "0" && !merge);
+    assert(parseMergeOpt("Y") == "1" && merge);
+    assert(parseMergeOpt("No") == "0" && !merge);
+    assert(parseMergeOpt("Yes") == "1" && merge);
+    assert(parseMergeOpt("NO") == "0" && !merge);
+    assert(parseMergeOpt("YES") == "1" && merge);
+    assert(parseMergeOpt("no") == "0" && !merge);
+    assert(parseMergeOpt("yes") == "1" && merge);
+    assert(parseMergeOpt("no") == "0" && !merge);
+    assert(parseMergeOpt("") == "1" && merge);
+    merge = false;
+    assert(parseMergeOpt("yEs") == "err" && !merge);
+    merge = true;
+    assert(parseMergeOpt("nO") == "err" && merge);
+    merge = false;
+    assert(parseMergeOpt("10") == "err" && !merge);
+    merge = true;
+    assert(parseMergeOpt("01") == "err" && merge);
+    assert(parseMergeOpt("xxx") == "err" && merge);
+    assert(parseMergeOpt("invalid") == "err" && merge);
+}
+
 shared static ~this()
 {
     if (!gdata.length) return;
 
+    // Parse options taken from the standard runtime options passing via
+    // --DRT-xxx / DRT_XXX.
+    import rt.config: rt_configOption;
+    rt_configOption("covmerge", (string opt) => parseMergeOpt(opt), false);
+    rt_configOption("covsrcpath", (string opt) => srcpath = opt, false);
+    rt_configOption("covdstpath", (string opt) => dstpath = opt, false);
```

Comment:
Adding a `static this()` wouldn't be much of a problem for an internal `rt.` module, but alternatively feel free to make the DRT parsing more modular and bootstrap it from the rt startup code.

Reply:
> You want to parse those options on startup instead of reporting an error after all tests have been run.

OK, fair point. There is still the ordering issue if someone decides to call the `dmd_cover*()` directly in some other module constructor, but it is a pretty obscure case and then if we could remove those functions in the future, I guess is not worth much to care about it.

Updated.